src/test/java/io/quarkus/search/app/fetching/FetchingServiceTest.java

```diff
@@ -252,6 +273,104 @@ void fetchQuarkusIo() throws Exception {
                                         JA_FETCHED_GUIDE_2_CONTENT_HTML));
             }
         }
+        // now let's update some guides and make sure that the content is fetched correctly:
+        updateMainRepository();
+
+        // NOTE that after an update we'll have non-translated titles and summaries,
+        //   since we've only updated them in the "main" repository,
```

Comment:
Better clarify this is not a limitation of your implementation, it's just how the test is designed :)

```suggestion
        //   since in this test we've only updated them in the "main" repository,
```